tests/nv_storage_nv_index_attributes_tests.rs

```diff
@@ -60,78 +63,289 @@ mod test_nv_storage_nv_index_attributes {
+    #[test]
+    fn test_nv_index_attributes_builder_missing_read_attribute_failure() {
+        // Test missing read error
+
+        // Building with only PP write set this should result in an error
+        single_attribute_error!(with_pp_write);
+        // Building with only owner write set this should result in an error
+        single_attribute_error!(with_owner_write);
+        // Building with only auth write set this should result in an error
+        single_attribute_error!(with_auth_write);
+        // Building with only policy write set this should result in an error
+        single_attribute_error!(with_policy_write);
+        // Building with all of them set still results in an error
+        assert_eq!(
+            Err(Error::WrapperError(WrapperErrorKind::ParamsMissing)),
+            NvIndexAttributesBuilder::new()
+                .with_pp_write(true)
+                .with_owner_write(true)
+                .with_auth_write(true)
+                .with_policy_write(true)
+                .build()
+        );
+    }
 
-        attributes.set_auth_write(true);
-        assert_eq!(0b0000_0000_0000_0000_0000_0000_0000_0111u32, attributes.0);
+    #[test]
+    fn test_nv_index_attributes_builder_missing_write_attribute_failure() {
+        // Test missing read error
+
+        // Building with only PP read set this should result in an error
+        single_attribute_error!(with_pp_read);
+        // Building with only owner read set this should result in an error
+        single_attribute_error!(with_owner_read);
+        // Building with only auth read set this should result in an error
+        single_attribute_error!(with_auth_read);
+        // Building with only policy read set this should result in an error
+        single_attribute_error!(with_policy_read);
+        // Building with all of them set still results in an error
+        assert_eq!(
+            Err(Error::WrapperError(WrapperErrorKind::ParamsMissing)),
+            NvIndexAttributesBuilder::new()
+                .with_pp_read(true)
+                .with_owner_read(true)
+                .with_auth_read(true)
+                .with_policy_read(true)
+                .build()
+        );
+    }
 
-        attributes.set_policy_write(true);
-        assert_eq!(0b0000_0000_0000_0000_0000_0000_0000_1111u32, attributes.0);
+    #[test]
+    fn test_nv_index_attributes_builder_missing_no_da_attribute_with_pin_fail_index_type() {
+        assert_eq!(
+            Err(Error::WrapperError(WrapperErrorKind::ParamsMissing)),
+            NvIndexAttributesBuilder::new()
+                .with_nv_index_type(NvIndexType::PinFail)
+                .build()
+        );
+    }
 
-        // PinPass = 1001 (7,4)
-        attributes.set_index_type(NvIndexType::PinPass);
-        assert_eq!(0b0000_0000_0000_0000_0000_0000_1001_1111u32, attributes.0);
-        assert_eq!(NvIndexType::PinPass, attributes.index_type().unwrap());
+    #[test]
+    fn test_attributes_builder() {
```

Comment:
I'm debating the usefulness of this test - I guess it's testing the builder more than the code in `bitfield` (which would've been a bit weird). But I'm happy to keep it 😄

Reply:
It sort of test the entire chain. From builder to attributes into raw value. So that the correct bit is set in the TSS value when you call corresponding function in builder with true as argument.